### PR TITLE
Fix broken links in book

### PR DESCRIPTION
**Pull request by Roms1383**
A couple of links are broken in the docs, see #731.
It fixes them and add [mdbook-linkcheck](https://github.com/Michael-F-Bryan/mdbook-linkcheck) to the CI.

## Checklist

- [x] An issue to be fixed by this PR is listed above.
- [ ] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [x] The code generator is run and the code is formatted (e.g. via `just refresh_all`).
- [ ] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [x] CI is passing.

### CODE REVIEW

File: .github/workflows/gh-pages.yaml [hunk 1]
```diff
@@ -25,8 +25,8 @@ jobs:
         with:
           mdbook-version: 'latest'
       
-      - name: Setup `mdbook-mermaid`
-        run: cargo install mdbook-mermaid
+      - name: Setup mdbook plugins (mermaid, linkcheck)
```
Comment (fzyzcjy) [inline .github/workflows/gh-pages.yaml:28]: Wonder whether this is enough to auto check in CI? If so then OK

**Review by fzyzcjy — APPROVED** [review]
LGTM % CI passes, good job :)

File: .github/workflows/gh-pages.yaml (hunk 1 above)
Comment (Roms1383) [reply .github/workflows/gh-pages.yaml:28]: Yes, it checks automatically when `mdbook build` runs (which is the command ran just after: `mdbook build ./book`).

**Comment by fzyzcjy** [comment]
Temporarily revert this to see whether #741 is ok now. Feel free to re-PR without this bug!

**Comment by Roms1383** [comment]
@fzyzcjy I browsed the files again there shouldn't be any problem : could you describe more accurately what kind of bugs it generated ?

**Comment by fzyzcjy** [comment]
@Roms1383 when viewing the website every page says 404

**Pull request merged**